provd/proto/ad.proto

```diff
@@ -0,0 +1,113 @@
+syntax = "proto3";
+
+package active_directory;
+
+service ActiveDirectoryService {
+    rpc HasSupport(Empty) returns (HasSupportResponse);
+    rpc IsUsed(Empty) returns (IsUsedResponse);
+    rpc SetUsed(SetUsedRequest) returns (Empty);
+    rpc GetConnectionInfo(Empty) returns (GetConnectionInfoResponse);
+    rpc SetConnectionInfo(SetConnectionInfoRequest) returns (Empty);
+    rpc CheckDomainName(CheckDomainNameRequest) returns (CheckDomainNameResponse);
+    rpc CheckAdminName(CheckAdminNameRequest) returns (CheckAdminNameResponse);
+    rpc CheckPassword(CheckPasswordRequest) returns (CheckPasswordResponse);
+    rpc PingDomainController(PingDomainControllerRequest) returns (PingDomainControllerResponse);
+    rpc GetJoinResult(GetJoinResultRequest) returns (GetJoinResultResponse);
```

Comment:
I suspect my questions/comments are because I don't know AD , but these method names could be improved. For example, "HasSupport", "Is Used" and "SetUsed" seem too generic to me and we can't assume the UI consuming this API will have deep AD knowledge. I have similar questions for many of the types here.

Lets jump on a call and discuss @matthew-hagemann?

Reply:
I'll do some digging on what all these settings actually do to see if we can improve the naming past what was used in the abstract classes :+1: